src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -195,11 +195,12 @@ namespace winrt::Microsoft::Terminal::Control::implementation
                 }
             });
 
-        _updatePatternLocations = std::make_shared<ThrottledFuncTrailing<>>(
-            _dispatcher,
+        // NOTE: Calling UpdatePatternLocations from a background
+        // thread is a workaround for us to hit GH#12607 less often.
+        _updatePatternLocations = std::make_unique<til::throttled_func_trailing<>>(
             UpdatePatternLocationsInterval,
             [weakThis = get_weak()]() {
-                if (auto core{ weakThis.get() }; !core->_IsClosing())
+                if (auto core{ weakThis.get() })

```

Comment:
How can we be sure now that the control isn't tearing down?

Reply:
We simply can't. After all, `_IsClosing` isn't a mutex which would prevent the main thread from closing under us while we're running on the background thread.
Thankfully, `Terminal::_InvalidateFromCoords` doesn't seem to depend on any UI state.